frontend/plugin_init.rb

```diff
@@ -13,6 +13,8 @@ def self.AGENT_FACETS
         def render_aspace_partial(args)
             if args[:partial] == "search/listing"
                 if controller.controller_name == 'agents' || ASUtils.wrap(@search_data['results']).any?{|result| ASUtils.wrap(result['types']).include?("agent")}
+                    add_columns
+
                     action_column = @columns.pop if @columns.last.class.include?('actions')
```

Comment:
The `last` call here seems to be the origin of the traceback. For whatever reason, `@columns` is defined when the user can CRUD, but not when they have view-only permissions. Adding this `add_columns` helper (see https://github.com/archivesspace/archivesspace/blob/889b8e8701326727f923c44a70fecdb8dc4664db/frontend/app/helpers/search_helper.rb#L301-L319) makes the page load, though I may not know enough about the application or plugin to know if things are working properly thereafter.

I'm still going to look into why CRUD users get `@columns`. My current hypothesis is that this template block (https://github.com/archivesspace/archivesspace/blob/889b8e8701326727f923c44a70fecdb8dc4664db/frontend/app/views/agents/index.html.erb#L10-L76) is adding `@columns` behind the scenes, since without it the template also breaks for CRUD users with the same error.

Based on my current understanding of the code here and Rails in general, I don't think the `add_columns` in `search/_listing.hrml.erb` is called yet, since that partial hasn't been rendered at this point. https://github.com/archivesspace/archivesspace/blob/889b8e8701326727f923c44a70fecdb8dc4664db/frontend/app/views/search/_listing.html.erb#L3

I'd be happy to have others correct my understanding here or weigh in.

Reply:
Okay, turns out it's the [`fields` method call here](https://github.com/archivesspace/archivesspace/blob/889b8e8701326727f923c44a70fecdb8dc4664db/frontend/app/views/agents/index.html.erb#L43).

This method sets `@columns` if it hasn't been set yet, see the [`fields` helper here](https://github.com/archivesspace/archivesspace/blob/889b8e8701326727f923c44a70fecdb8dc4664db/frontend/app/helpers/search_helper.rb#L378-L381).